### PR TITLE
[WIP] Adds support for binary requests (application/octet-stream, multipart/form-data)

### PR DESCRIPTION
**Pull request by ryanblock**
Thank you for helping out! ✨

Before submitting a pull request, please make sure the you checked the following:

- [x] Forked the repository and created your branch from master
- [x] Make sure the tests pass! `npm it` in the repository root
- [x] If you've fixed a bug or added code **more** tests are appreciated
- [x] If you're adding a new command, removing an old command, or changing how a command works, please update the relevant documentation for the command (under the `doc/` folder)
- [x] Summarize your changes in the [changelog](https://github.com/arc-repos/architect/blob/master/changelog.md)
- [x] If you haven't already please complete the CLA

Learn more about contributing: https://arc.codes/intro/community

### CODE REVIEW

File: src/create/aws/create-http-route/create-route/02-setup-request.js [hunk 1]
```diff
@@ -56,13 +57,14 @@ module.exports = function _02setupRequest(params, callback) {
             integrationHttpMethod: 'POST',
             uri,
             requestTemplates: {
-              'text/html': vtl,
-              'application/x-www-form-urlencoded': vtlForm,
-              'multipart/form-data': vtl,
               'application/json': vtl,
+              'application/octet-stream': vtlBinary,
```
Comment (mikeal) [inline src/create/aws/create-http-route/create-route/02-setup-request.js]: At some point you may also want to add other known binary content types. You’re going to want the same thing for pretty much every audio, video and image (except SVG) mime type.

**Comment by mikeal** [comment]
Looking through these vtl diffs gave me a new appreciation of the work arc is doing for me :)

File: src/create/aws/create-http-route/create-route/02-setup-request.js (hunk 1 above)
Comment (ryanblock) [reply src/create/aws/create-http-route/create-route/02-setup-request.js]: Right! Depends on how we want to approach that. On the one hand, we could tell people to just upload with `application/octet-stream` or encode with `multipart/form-data`.

Could also add templates for every common binary audio, video, and image format, which feels noisy. Sadly, we can't (yet) do something like `image/*`.

**Comment by ryanblock** [comment]
@mikeal yeah it gets real gnarly in there. And unfortunately, API Gateway is extra challenging: no dev environment, basically a black box, nearly useless documentation, and all kinds of weird encoding rules on ingress and egress. I feel we've done as good a job as anyone at taming it, but there is still much work to be done!

**Pull request merged**